containers/app/Dockerfile

```diff
@@ -32,21 +32,32 @@ FROM python:3.12-slim as runtime
 
 WORKDIR /app
 
-ENV RUN_AS_DEVIN=false
-ENV USE_HOST_NETWORK=false
+ENV RUN_AS_DEVIN=true
+ENV SANDBOX_USER_ID=1000
+ENV USE_HOST_NETWORK=true
 ENV SSH_HOSTNAME=host.docker.internal
 ENV WORKSPACE_BASE=/opt/workspace_base
 ENV OPEN_DEVIN_BUILD_VERSION=$OPEN_DEVIN_BUILD_VERSION
 RUN mkdir -p $WORKSPACE_BASE
 
 RUN apt-get update -y \
-    && apt-get install -y curl ssh
+    && apt-get install -y curl ssh sudo
+
+RUN useradd -m -u $SANDBOX_USER_ID -s /bin/bash opendevin && \
+    usermod -aG sudo opendevin && \
+    echo '%sudo ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers
+RUN chown -R opendevin:opendevin /app
+USER opendevin
```

Comment:
Can you explain why the sandbox user is in the app container? That doesn't seem right to me....

Reply:
This is something i did in https://github.com/OpenDevin/OpenDevin/pull/1426

maybe we can merge this first. The reason is that, when mounted dir in DooD setup, if the user is `root` but not `SANDBOX_USER_ID`, there's a chance that the `app` server will encounter permission denied error in writing to `/opt/workspace_base`.

Reply:
👍 thanks for the explanation! It does make sense. I hope we figure out a way to do this better in the future but this definitely makes sense for now